internal/compiler/generator/cpp.rs

```diff
@@ -1961,6 +1961,30 @@ fn generate_public_api_for_properties(
                 ]),
                 ..Default::default()
             }));
+        } else if let Type::Function { return_type, args } = &p.ty {
+            let param_types = args.iter().map(|t| t.cpp_type().unwrap()).collect::<Vec<_>>();
+            let ret = return_type.cpp_type().unwrap();
```

Comment:
For callbacks there's an extra dance for `void` - I guess the code there could be simplified in the same way as you're unwrapping here because `cpp_type()` supports void?

Reply:
The callback uses an `Option<Type>`  for the return type, but the function use Type::Void 
Arguably it could be merged